front/src/components/ChangeNamePopup.js

```diff
@@ -0,0 +1,76 @@
+import React, {Component} from "react";
+import AuthenticationService from "../service/AuthenticationService";
+import axios from "axios";
+import {Button, Form, FormGroup, Input} from "reactstrap";
+
+class ChangeNamePopup extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            serverUrl: "http://localhost:8080",
+            projectId: this.props.id,
+            projectName: "",
+            showPopup: false
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+
+        let headers = {headers: {authorization: AuthenticationService.getAuthToken()}};
+        axios.get(this.state.serverUrl + '/projects/id/' + this.state.projectId,headers)
+            .then((response) => {
+                const data = response.data;
+                this.setState({projectName: data.name});
+            })
+            .catch(response => {
+                console.log("Error: " + response);
+            });
+    }
+
+    handleChange(event) {
+        const {name, value} = event.target;
+        this.setState({[name]: value});
+    }
+
+    submitForm = (e) => {
+        e.preventDefault();
+        const url = this.state.serverUrl + "/projects/changename/" + this.state.projectId;
+        axios.post(url, null, {params: this.state, headers: {authentication: AuthenticationService.getAuthToken()}})
```

Comment:
- `authentication` field in headers should be changed to `authorization` - currently it returns 401 error
- it is not necessary to send the whole state - `this.state.projectName` would be enough